llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2448,6 +2448,16 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
     }
   }
 
+  {
+    // (sub (xor X, (sext C)), (sext C)) => (select C (neg X), X)
+    Value *C0, *C1, *X;
+    if (match(Op0, m_Xor(m_Value(X), m_SExt(m_Value(C0)))) &&
```

Comment:
Please also handle the commuted version: `(sub (sext C), (xor X, (sext C))) -> select C, X, (neg X)`.

Missing one-use checks?